redpanda-data/tinygo
# Update upstream tinygo to release 29

- goenv: update to new v0.29.0 development version
- ci: rename release-double-zipped to something more useful
- os: define ErrNoDeadline
- reflect: use .key() instead of a type assert
- ci: update Node.js from version 14 to version 16
- wasm: remove i64 workaround, use BigInt instead
- example: simplify pininterrupt
- nrf: wait for stop condition after reading from the I2C bus
- machine/usb/hid,joystick: fix hidreport (3) (#3802)
- reflect: remove unecessary heap allocations
- reflect: Add FieldByNameFunc
- docker: update clang to version 15
- reflect: fix iterating over maps with interface{} keys
- compiler,reflect: NumMethods reports exported methods only
- compiler: update .ll test output
- example: adjust time offset
- rp2040: add missing suffix to CMD_READ_STATUS
- machine/macropad_rp2040: add machine.BUTTON
- main: improve detection of filesystems
- modules: add submodule for Renesas SVD file mirror repo
- tools/gen-device-svd: small changes needed for Renesas MCUs
- make: add make task to generate Renesas device wrappers
- machine/samd51: fix i2cTimeout was decreasing due to cache activation
- reflect: implement Value.Grow
- wasi: allow zero inodes when reading directories
- main: use `go env` instead of doing all detection manually
- reflect: add SetZero
- machine/rp2040: wait for 1000 us after flash reset to avoid issues with busy USB bus
- machine.UART refactor (#3832)
- nrf: add I2C timeout
- rp2040:add NoPin support
- sync: add implementation from upstream Go for OnceFunc, OnceValue, and OnceValues
- avr: pin change interrupt
- wasm: fix functions exported through //export
- compiler: add compiler-rt and wasm symbols to table
- compiler: add compiler-rt to wasm.json
- machine/usb: remove usbDescriptorConfig
- machine/usb: allow USB Endpoint settings to be changed externally
- machine/usb: refactor endpoint configuration
- machine/usb/hid: add RxHandler interface
- machine/usb/hid: rename Handler() to TxHandler()
- testing: add Testing function
- all: Go 1.21 support
- compiler: add min and max builtin support
- compiler: implement clear builtin for slices
- compiler: implement clear builtin for maps
- compiler: improve panic message when a runtime call is unavailable
- board: add AKIZUKI DENSHI AE-RP2040
- all: use https for renesas submodule #3856
- Add support for HID Keyboard LEDs
- builder: update message for max supported Go version
- build: switch GH actions builds to use Go 1.21 final release
- docker: use Go 1.21 for Docker dev container build
- modules: update to go-serial package v1.6.0
- rp2040: move flash related functions into separate file from C imports for correct LSP. Fixes #3852
- main: add target JSON file in `tinygo info` output
- loader: merge go.env file which is now required starting in Go 1.21 to correctly get required packages
- samd51,rp2040,nrf528xx,stm32: implement watchdog
- wasm: add support for GOOS=wasip1
- adding new uf2 target for PCA10056 (#3765)
- make: add task to check NodeJS version before running tests
- nRF52: set SPI TX/RX lengths even data is empty. Fixes #3868 (#3877)
- all: update version for 0.29 release
- docs: update CHANGELOG for release 0.29
- Repackage tinygo as an RPK plugin
- Add autocomplete commands
- Smaller release tar
- Resolve symlinks when looking for TINYGOROOT
- Update build command
- Package binarygen on macos
- Add help text after builds
- Repackage tinygo as an RPK plugin
- Revert "Resolve symlinks when looking for TINYGOROOT"
- Revert "Add autocomplete commands"
- Revert "Repackage tinygo as an RPK plugin"
